### ISSUE TITLE
CSP Violation

### ISSUE THREAD

**Issue by sheremet-va** [XiongAmao/vue-easy-lightbox#75]
![Снимок экрана 2021-08-02 в 15 20 44](https://user-images.githubusercontent.com/16173870/127861022-c688fe7f-ee30-4ff4-949e-e419080eecb9.png)
<img width="1045" alt="Снимок экрана 2021-08-02 в 15 23 03" src="https://user-images.githubusercontent.com/16173870/127861292-3bd6170f-e7f7-4c1d-a71f-ba5796cbcb69.png">

This package is inlining css, which violates my CSP rules (`default-src 'self'`). I know I can enable `unsafe-inline` but this is unsafe and kinda defeats the purpose.

Is it possible to bundle this package in some other way?

**Comment by XiongAmao** [comment]
Sorry for the late reply.

I have released v1.2.2, which may fix this issue.
I extract css to `dist/external-css/vue-easy-lightbox.css`. And .js in this directory do not contain inlining css.

```js
// in this path vue-easy-lightbox/dist/external-css/*.js 
import VueEasyLightbox from 'vue-easy-lightbox/dist/external-css/vue-easy-lightbox.es5.esm.min.js'

// you need to import css yourself
import 'vue-easy-lightbox/external-css/vue-easy-lightbox.css' 
```

**Comment by sheremet-va** [comment]
Hey! Thank you very much for the clean fix!

However with typescript importing works, but gives me the error:
![Снимок экрана 2021-08-18 в 15 19 23](https://user-images.githubusercontent.com/16173870/129896881-fb6480ec-331f-4072-af35-9183d45170fd.png)

For now I fixed it locally with:
```
declare module 'vue-easy-lightbox/dist/external-css/vue-easy-lightbox.es5.esm.min.js';
```

But I'm not sure, since the package has `types.d.ts`, so TypeScript support is expected.

**Comment by XiongAmao** [comment]
I did a search. But haven't found how to add `d.ts` for named module path `dist/*.js` yet.

As you mentioned, add `d.ts` locally is one way: 
```ts
declare module 'vue-easy-lightbox/dist/external-css/vue-easy-lightbox.es5.common.min' {
  import VueEasyLightbox from 'vue-easy-lightbox'
  export default VueEasyLightbox
}
```

The other way, if you're using webpack: 
https://webpack.js.org/configuration/resolve/#resolvealias

```js
// wepback.config.js
module.exports = {
  //...
  resolve: {
    alias: {
      'vue-easy-lightbox$': 'vue-easy-lightbox/dist/external-css/vue-easy-lightbox.es5.common.min.js',
    },
  },
};

// in your component
import VueEasyLightbox from 'vue-easy-lightbox' // work
```

**Comment by sheremet-va** [comment]
Thank you for your advice with aliasing. Will do that. 

I think it can be added to the docs.

**Comment by XiongAmao** [comment]
Yes, docs updated. 
https://onycat.com/vue-easy-lightbox/guide/#external-css-build-1-2-3

**Comment by sheremet-va** [comment]
Thank you! Closing the issue now

**Comment by ekrausso** [comment]
Hi @XiongAmao are external-css builds available for vue2 component or only vue3? thanks

**Comment by XiongAmao** [comment]
@ekrausso 
Hi, it is released in v0.19.0.

**Comment by ekrausso** [comment]
@XiongAmao great!! thanks :)
